internal/jobserver/client/env.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package client
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/datadog/orchestrion/internal/filelock"
+	"github.com/datadog/orchestrion/internal/log"
+)
+
+const (
+	ENV_VAR_JOBSERVER_URL = "ORCHESTRION_JOBSERVER_URL"
+	urlFileName           = ".orchestrion-jobserver"
+)
+
+var (
+	client *Client
+
+	ErrNoServerAvailable = errors.New("no job server is available")
+)
+
+// FromEnvironment returns a client connected to the current environment's
+// job server, using the following process:
+//   - If the ORCHESTRION_JOBSERVER_URL environment variable is set, a client
+//     connected to this URL is returned.
+//   - Otherwise, if workDir is not empty, a server will be identified based on
+//     a `.orchestrion-jobserver` file; or a new server will be started using
+//     that url file, and a connection will be established to it. The started
+//     job server will automatically shut itself down once it no longer has any
+//     active client for a period of time.
+//   - Otherwise, the ErrNoServerAvailable error is returned.
+func FromEnvironment(workDir string) (*Client, error) {
+	if client == nil {
```

Comment:
Can you revert this if statement and make it a returning-if to lower tabulation levers?